### ISSUE TITLE
Bug: Ctrl+Drag&Drop (copy) a selection beginning with `#` creates a bad link

### ISSUE THREAD

**Issue by jbridgy** [RickStrahl/MarkdownMonster#1002]
MM v2.7.14 always creates a bad link when you Ctrl+Drag&Drop (copy) a selection that begins with a '#' character.

Creating a correct link (as with Ctrl+K dialog) would be very welcome if the selection is an entire header line,
otherwise the selection should be copied normally.

When a link is created then the selection without leading '#' and ' ' characters and without end of line should be used as `Link Text`.

EXAMPLE:

MM created the following link when the selection contained the text above beginning with "'#' character":

```markdown
[' character.

Creating a correct link (as with Ctrl+K dialog) would be very welcome if the selection is an entire header line,
otherwise the selection should be copied normally.

When a link is created then the selection without leading '#' and ' ' characters and without end of line should be used as `Link Text`.
](#' character.

Creating a correct link (as with Ctrl+K dialog) would be very welcome if the selection is an entire header line,
otherwise the selection should be copied normally.

When a link is created then the selection without leading '#' and ' ' characters and without end of line should be used as `Link Text`.
)
```

**Comment by RickStrahl** [comment]
Do you mean editor to editor selection drag and drop?

Can you post the text you're dragging? I assume the above is the result of the drop?

**Comment by jbridgy** [comment]
> Do you mean editor to editor selection drag and drop?

I mean primarily Ctrl+Drag&Drop a selection beginning with a `#` character within the main editor of MM. However you get the very same result when the selection comes from another editor or a web page.
 
> Can you post the text you're dragging? I assume the above is the result of the drop?

It is already there. The beginning of my initial post is used as source of the drag. Here it is again as proper Markdown text (which is in this case almost identical to the rendered text):

```markdown
MM v2.7.14 always creates a bad link when you Ctrl+Drag&Drop (copy) a selection that begins with a '#' character.

Creating a correct link (as with Ctrl+K dialog) would be very welcome if the selection is an entire header line,
otherwise the selection should be copied normally.

When a link is created then the selection without leading '#' and ' ' characters and without end of line should be used as `Link Text`.
```

The selection started with the `#` character immediately after `MM v2.7.14 always creates a bad link when you Ctrl+Drag&Drop (copy) a selection that begins with a '`.

**Comment by RickStrahl** [comment]
You posted what got pasted not what was dragged.

**Comment by jbridgy** [comment]
Are you joking? Already the title gives enough information to reproduce the bug. The rest just elaborates on that and uses the post itself as an example (kind of recursive).

Here is another example:
1. Create a new "untitled" document in MM with the following text:

```markdown
Bug: Ctrl+Drag&Drop (copy) a selection beginning with # creates a bad link
Another non-empty line to demonstrate the bug more obviously.

Drop here:

Last non-empty line.
```

3. Select a part of the "untitled" document as shown in the following screenshot:
![image](https://user-images.githubusercontent.com/98041047/215915764-c2714fe1-55f3-4a1d-866c-3e963d26b732.png)


4. Ctrl+Drag&Drop the selection to the end of `Drop here:`. The result will be:
```markdown
Bug: Ctrl+Drag&Drop (copy) a selection beginning with # creates a bad link
Another non-empty line to demonstrate the bug more obviously.

Drop here:[ creates a bad link
Another non-empty line to demonstrate the bug more obviously.](# creates a bad link
Another non-empty line to demonstrate the bug more obviously.)

Last non-empty line.
```

Don't forget the **suggestion** embedded  in the initial post which reads:
1. Creating a correct link (as with Ctrl+K dialog) would be very welcome if the selection is an entire header line,
otherwise the selection should be copied normally.
2. When a link is created then the selection without leading `#` and ` ` (space) characters and without end of line should be used as `Link Text` (first part enclosed in \[backets\]).

Example:
![image](https://user-images.githubusercontent.com/98041047/215918640-ac25b393-628a-48d8-b00e-6a285c46fdaa.png)

Ctrl+Drag&Drop the selected header line to the end of `See also: ` should result in the following:
![image](https://user-images.githubusercontent.com/98041047/215918910-e2607f1b-b6b2-4a61-9c0b-aa6d8e353d04.png)

This result can be achieved already with the Ctrl+K dialog using the `#` button but less efficiently:
![image](https://user-images.githubusercontent.com/98041047/215919388-28d375e5-d4fe-4059-adf2-769e7c7c3918.png)

**Comment by RickStrahl** [comment]
The issue here is that when the link comes in there's no way to tell about context. The logic that happens here handles all document drop operations (coming from the WebBrowser interface) so it handles editor text drops as well as file drops and other things like bookmark link drops. Additionally the editor handles the drag and drop operation based on the key combos (ie. move by default, copy for ctrl-drag) which accounts for the original text being deleted or staying behind. There's not a lot of wiggle room to provide custom behavior here.

The link expansion was added initially for the bookmark drag feature that allows dragging bookmarks into the document. Luckily those bookmarks come in a specific format (ie.`#hash|linkText` so I can at least differentiate that). 

So, the best solution for this to simply not handle `#hash` drags specially and handle the specific case of the bookmark drops for link expansion. I've made the change, so now when you drag text that starts with # it just moves/copies the text without any special treatment. 

This is also inline with initial intent which never meant to auto-expand links in the first place.

It'll be in 2.8.3.1
